docs/troubleshoot/app-framework/appserver-error-codes.md

```diff
@@ -1800,291 +1800,344 @@ The following error message codes may appear on the app-server log. Use the foll
+  Could not extract war for service=_key-value_, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If the service with the _key-value_ pair is unable to extract the war file then it throws the _errorMessage_
 
   **Action:**
 
-  TODO
+  Check if the war file exists and configured correctly.
 
 
 
-### ZWED0078W":"Could not access files to determine status for service=%s, error=
+### ZWED0078W
+
+  Could not access files to determine status for service=_key-value_, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If we are unable to get the status of war extracted or not, then it throws _errorMessage_ in catch block.
 
   **Action:**
 
-  TODO
+  Check if the war file exists.
+
 
 
+### ZWED0079W
 
-### ZWED0079W":"Cannot add servlet for service=%s, error=
+  Cannot add servlet for service=_key-value_, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If unable to add servlet for service _key-value_, then it logs a warning _errorMessage_.
 
   **Action:**
 
-  TODO
+  No action is needed, but the warning may be needed to debug this War
+
 
 
+### ZWED0080W
 
-### ZWED0080W":"Cannot add servlet for service=%s
+  Cannot add servlet for service=_key-value_
 
   **Reason:**
 
-  TODO
+  When we are not able to get the _dir_ to add servlet for service _key-value_.
 
   **Action:**
 
-  TODO
+  Check if your _dir_ is exists and is valid.
 
 
 
-### ZWED0081W":"Could not start tomcat, error=
+### ZWED0081W
+
+  Could not start tomcat, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  Tomcat manager is unable to start tomcat with _DOptionsArray_.
 
   **Action:**
 
-  TODO
+  Review the warning message and verify _DOptionsArray_ element values are valid or not.
+
 
 
+### ZWED0082W
 
-### ZWED0082W":"%s stderr=%s
+  Tomcat PID=_pid_: stderr=_data_
 
   **Reason:**
 
-  TODO
+  A tomcat process with Tomcat _pid_ encountered an standard error with receiving _data_.
 
   **Action:**
 
-  TODO
+  Action depends on what _data_ is. May be useful in debugging.
+
 
 
+### ZWED0083W
 
-### ZWED0083W":"Tomcat could not start. Closing. code=%s
+  Tomcat could not start. Closing. code=_code_
 
   **Reason:**
 
-  TODO
+  If Tomcat Manager is unable to start Tomcat, then it closes the Tomcat.
 
   **Action:**
 
-  TODO
+  Review the warning message and see if you are able to check why the tomcat unable to start.
 
 
 
-### ZWED0084W":"Tomcat could not start. Exiting. code=%s
+### ZWED0084W
+
+  Tomcat could not start. Exiting. code=_code_
 
   **Reason:**
 
-  TODO
+  If Tomcat Manager is unable to start Tomcat, then it exists the Tomcat.
 
   **Action:**
 
-  TODO
+  Review the warning message and see if you are able to check why the tomcat unable to start
+  and exited from tomcat process.
+
 
 
+### ZWED0085
 
-### ZWED0085W":"%s Error when stopping, error=
+  Tomcat PID=_pid_ Error when stopping, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If Tomcat manager unable to stop tomcat-process for Windows then it logs _errorMessage_.
```

Comment:
If Tomcat manager is unable to stop the Tomcat process on Windows, then it [...]